lib/vsc/administration/ldapsync.py

```diff
@@ -177,34 +177,44 @@ def sync_altered_groups(self, last, dry_run=True):
         }
 
         for group in changed_groups:
-            vo = False
-            try:
-                vo = mkVo(self.client.vo[group.vsc_id].get()[1])
-            except HTTPError as err:
-                # if a 404 occured, the group is not an VO, so we skip this. Otherwise something else went wrong.
-                if err.code != 404:
-                    raise
+            # General group attributes
             group_moderators = [str(m) for m in group.moderators]
             institute_name = str(group.institute['name'])
-            if not group_moderators:
-                group_moderators = [str(VSC_CONFIG.backup_group_mods[group.institute['name']])]
-                logging.info("Using backup moderator %s for group %s", group_moderators, group.vsc_id)
+
             ldap_attributes = {
                 'cn': str(group.vsc_id),
                 'institute': [institute_name],
                 'gidNumber': ["%d" % (group.vsc_id_number,)],
                 'moderator': group_moderators,
-                'memberUid': [str(a) for a in group.members],
                 'status': [str(group.status)],
             }
-            if vo:
+
+            # Only set memberUid if there are actually active members in the group
+            # LDAP addition of records with empty memberUid will fail
+            if group.members:
+                ldap_attributes['memberUid'] = [str(a) for a in group.members]
+
+            # VO attributes
+            try:
+                vo = mkVo(self.client.vo[group.vsc_id].get()[1])
+            except HTTPError as err:
+                # if a 404 occured, the group is not an VO, so we skip this. Otherwise something else went wrong.
+                if err.code != 404:
+                    logger.raiseException("Retrieval of group VO failed for unexpected reasons")
+            else:
+                # Group is a VO
                 ldap_attributes['fairshare'] = ["%d" % (vo.fairshare,)]
                 ldap_attributes['description'] = [str(vo.description)]
                 ldap_attributes['dataDirectory'] = [str(vo.data_path)]
                 ldap_attributes['scratchDirectory'] = [str(vo.scratch_path)]
-                # vsc40024 is moderator for all institute vo's
-                if vo.vsc_id in INSTITUTE_VOS_GENT.values():
-                    ldap_attributes['moderator'] = ['vsc40024']
+                # Set institute moderator for main VOs
+                if vo.vsc_id in DEFAULT_VOS_ALL.values():
```

Comment:
`DEFAULT_VOS_ALL` is already a list, no need values()

Reply:
fixed